src/map/script.cpp

```diff
@@ -17234,6 +17234,88 @@ BUILDIN_FUNC(pcblockskill)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+BUILDIN_FUNC(pcblock)
+{
+	TBL_PC *sd;
+
+	if (!script_mapid2sd(4, sd))
+		return SCRIPT_CMD_SUCCESS;
+
+	enum pcblock_action_flag type = (pcblock_action_flag)script_getnum(st, 2);
+	int state = (script_getnum(st, 3) > 0) ? 1 : 0;
```

Comment:
state could be defined as bool.

```suggestion
	bool state = (script_getnum(st, 3) > 0) ? true : false;
```